### ISSUE TITLE
What is this, and how does it work now?

### ISSUE THREAD

**Issue by tek0011** [tek0011/reddit_info_bot#73]
Couldnt quite figure it out with the new changes.

https://github.com/tek0011/reddit_info_bot/blob/a91249bfae0bf73fb97d83270b3c36f6927db9f4/reddit_info_bot/__init__.py#L34-L39

**Comment by nyov** [comment]
That looks a bit convoluted, I acknowledge. In essence thats a case/switch statement using bit syntax; multiple options are assigned a number and encoded using bit operations into a number, that number is then later decoded to decide which options are active.
So in the config, that option is a list now, to allow adding multiple things there.

It doesn't make much sense here now, there's only one mode active usually (log, pm or comment) but I was considering having more or allowing them in parallel (logging and replying, mailing the output, writing to a gist, or such - which would be non-exclusive options then).

**Comment by tek0011** [comment]
I understand what the code is doing, sorry.

What Im trying to remember is what those settings even do lol.  I think its always been in PM mode.  Dont really remember what I used all those other settings for

**Comment by nyov** [comment]
In 'log' mode it only prints results from the bot actions, in 'comment' mode it replies to requests/comments as would be expected. 'pm' mode is a bit strange, it only differs for the keyword searches. There it replies to "potential comments" that have any of the keywords in `BOTCMD_INFORMATIONAL` with a pm using the `INFOREPLY_MESSAGE` -- in 'comment' mode it would reply on the comment itself (with a public comment).

The whole thing looks a bit grown, i'll think on how this could be better implemented (probably remove this as a 'global' setting, split it up).

**Comment by tek0011** [comment]
ahh ok.  thanks for the explanation.  i remember what the settings do now.
 ya they might be a bit stale and repetitive.

what is your paypal please?

On Fri, Oct 23, 2015 at 9:29 PM, nyov <notifications@github.com> wrote:

> In 'log' mode it only prints results from the bot actions, in 'comment'
> mode it replies to requests/comments as would be expected. 'pm' mode is a
> bit strange, it only differs for the keyword searches. There it replies to
> "potential comments" that have any of the keywords in BOTCMD_INFORMATIONAL
> with a pm using the INFOREPLY_MESSAGE -- in 'comment' mode it would reply
> on the comment itself (with a public comment).
>
> The whole thing looks a bit grown, i'll think on how this could be better
> implemented (probably remove this as a 'global' setting, split it up).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tek0011/reddit_info_bot/issues/73#issuecomment-150742754>
> .
>



-- 
Jeff
